esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2252,38 +2252,46 @@ void CFileSprayEx::getDropZoneInfoByIP(double clientVersion, const char* ip, con
         return;
     }
 
-    IArrayOf<IConstTpDropZone> dropZones;
-    queryDropZoneInfo(clientVersion, nullptr, false, dropZones);
-    if (dropZones.empty())
-        throw MakeStringException(ECLWATCH_DROP_ZONE_NOT_FOUND, "No dropzone found.");
-
     bool dzFound = false;
     IpAddress ipAddrToMatch(ip);
-    ForEachItemIn(currentDZIndex, dropZones)
+
+    Owned<IConstDropZoneInfoIterator> dropZoneItr = constEnv->getDropZoneIterator();
+    ForEach(*dropZoneItr)
     {
-        IConstTpDropZone& dz = dropZones.item(currentDZIndex);
-        IArrayOf<IConstTpMachine>& servers = dz.getTpMachines();
-        ForEachItemIn(currentServerIndex, servers)
+        IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
+
+        SCMStringBuffer dropZoneDirectory, dropZoneUMask;
+        dropZoneInfo.getDirectory(dropZoneDirectory);
+        dropZoneInfo.getUMask(dropZoneUMask);
+        if (!dropZoneDirectory.length())
+            continue;
+
+        Owned<IConstDropZoneServerInfoIterator> machineItr = dropZoneInfo.getServers();
```

Comment:
The rest of this method would be a lot cleaner/more readable if a iterator for DropZone's that matched IP were encapsulated and handled here (i.e. what you requested in HPCC-17307)
And since that's effectively what you're not only coding here, but also as part of checkDropZoneIPAndPath/checkDropZoneIPAddressAndPath, it would a much better to actually implement a IConstDropZoneInfoIterator *getDropZonesByIP()  and use it, the code would look a lot simpler.